### PR TITLE
add file count ability

### PR DESCRIPTION
**Pull request by dsschult**

### CODE REVIEW

File: file_catalog/server.py [hunk 1]
```diff
@@ -408,7 +408,10 @@ def get(self):
             logging.warn('query parameter error', exc_info=True)
             self.send_error(400, message='invalid query parameters')
             return
-        files = yield self.db.find_files(**kwargs)
+        if 'count' in kwargs:
+            files = yield self.db.count_files(**kwargs)
```
Comment (ric-evans) [inline file_catalog/server.py:412]: I don't think that dynamically changing the type of `files` is intuitive. I expected `files` to always be a list.

File: tests/test_files.py [hunk 2]
```diff
@@ -44,6 +44,30 @@ def test_10_files(self):
...
+            u'locations': [{u'site':u'test',u'path':u'blah.dat'}]
+        }
+        data = r.request_seq('POST', '/api/files', metadata)
+        self.assertIn('_links', data)
+        self.assertIn('self', data['_links'])
+        self.assertIn('file', data)
+        url = data['file']
+        uid = url.split('/')[-1]
+
+        data = r.request_seq('GET', '/api/files?count=1')
```
Comment (blinkdog) [inline tests/test_files.py:65]: Per discussion in #dev-sprint, let's make this `/api/files/count?query=...`

Comment (dsschult) [reply tests/test_files.py]: And done.

**Comment by blinkdog** [comment]
I like the refactoring of the query building.

**Pull request merged**